TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/SourceTargetCaptureTuple.java

```diff
@@ -119,6 +125,26 @@ private JSONObject toJSONObject(SourceTargetCaptureTuple tuple) {
                     .ifPresent(d-> meta.put("targetResponse", jsonFromHttpData(d, tuple.targetResponseDuration)));
             meta.put("connectionId", tuple.uniqueRequestKey);
             Optional.ofNullable(tuple.errorCause).ifPresent(e->meta.put("error", e));
+
+            if (meta.has("sourceResponse") && meta.has("targetResponse")) {
+                String sourceResponseStatus = (String) meta.getJSONObject("sourceResponse").get("Status-Code");
+                String targetResponseStatus = (String) meta.getJSONObject("targetResponse").get("Status-Code");
```

Comment:
The sourceResponse/targetResponse are filled in with code that runs through this function...
```
        private JSONObject jsonFromHttpData(@NonNull List<byte[]> data) {
            try {
                return jsonFromHttpDataUnsafe(data);
            } catch (Exception e) {
                log.warn("Putting what may be a bogus value in the output because transforming it " +
                        "into json threw an exception");
                return new JSONObject(Map.of("Exception", e.toString()));
            }
        }
```

That means that we might have the "Status-Code" values of null.  That will be an issue down at line 137 where we'll throw an NPE.